src/codeu/chat/common/ServerInfo.java

```diff
@@ -0,0 +1,31 @@
+
+package codeu.chat.common;
+
+import codeu.chat.util.Uuid;
+import java.io.IOException;
+
+public final class ServerInfo {
+  private final static String SERVER_VERSION = "1.0.0";
+  public final Time startTime;
+
+  public final Uuid version;
+  private Uuid parseResult;
+
+  public ServerInfo() {
+    try {
+    	this.startTime = Time.now();
+      parseResult = Uuid.parse(SERVER_VERSION);
+    } catch (IOException exception) {
+      exception.getMessage();
+    }
+    version = parseResult;
+  }
+
+  public ServerInfo(Uuid version) {
+    this.version = version;
+  }
+  public ServerInfo(Time startTime){
```

Comment:
These constructors should be giving you a compilation error. You can't just set one field, they are both `final`. If you wanted a constructor where you set them you would need to pass in both fields, i.e. `public ServerInfo(Uuid version, Time startTime) {`.

But for that to work you will need to have the version sent from the server to the client, which means you will need to incorporate all the rest of the Version check code. It's good you tried to here, but it won't work properly unless you do it with the rest of the code. I would change this as if `version` didn't exist and only add it in when we merge that PR in.